app/models/mixins/relationship_mixin.rb

```diff
@@ -109,6 +110,19 @@ def relationship_ids
   # has_ancestry methods
   #
 
+  # Returns the id in the relationship table for this record's parents
+  # from this id, relationship records can be brought back and mapped to the resource of interest
+  # NOTE: parent_id is read from ancestry field, so it is not a db hit, but parent is an N+1 db hit.
+  def relationship_parent_ids
+    relationships.where.not(:ancestry => [nil, ""]).select(:ancestry).collect(&:parent_id)
+  end
+
+  # Returns relationships for the root relationships
+  # NOTE: can't cache this since it is a scope. caching relationship_parent_ids instead
+  def rel_roots(resource_type = {})
```

Comment:
I don't quite understand what this is looking up...can you explain?  roots are just `.roots`, right?  or given a relationship, just call `.root`  What I don't understand is what roots have to do with parents.